doc/sai-ptf/lag_test_plan.md

```diff
@@ -0,0 +1,140 @@
+sai_thrift_fdb_entry_t(switch_id=self.switch_id, mac_address=mac1, bv_id=self.vlan_oid)
+sai_thrift_create_fdb_entry(
+            self.client,
+            fdb_entry,
+            type=SAI_FDB_ENTRY_TYPE_STATIC,
+            bridge_port_id=port_bp,
+            packet_action=SAI_PACKET_ACTION_FORWARD)
+```
+
+Remove lag member
+```Python
+sai_thrift_remove_lag_member(self.client,  self.lag1_member6)
+```
+
+## Packets
+Vlan tagged packet
+```Python
+simple_udp_packet(eth_dst=dst_mac,
+                    eth_src=src_mac,
+                    dl_vlan_enable=True,
+                    vlan_vid=vlan_id,
+                    pktlen=104)
+```
+
+# Test suites
+## Test suite #1: PortChannel Loadbalanceing
+For load balancing, expecting the ports in a lag should receive the packet equally.
+
+Even after removing and disabling the port in a lag.
+
+Sample APIS
+Disbale egress
+```Python
+sai_thrift_set_lag_member_attribute(
+                self.client,
+                self.lag1_member4,
+                ingress_disable=True,
+                egress_disable=True)
+```
+
+| Steps/Cases | Goal | Expect  |
+|-|-|-|
+| Create lag, add lag members, port1 - 4. Add FDB entry for lag, map with a MAC. | Create lag and member| lag, and member created|
+| Send packet with.| Prepare to send from lag to VLAN.| Vlan and members have been created.|
+| Send packet on port0 to the lag by specifying lag mac as dest mac. 4 times .| Packet forwards on port equally.| Loadbalance on lag members.|
+| Every time, disable egress/ingress on one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+| Every time, enable egress/ingress on one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+| Every time, remove one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+
+## Test suite #2: Ingress/Egreee disable
+Sample APIs
+
+Ingress/Egreee disable
+```python
+    status = sai_thrift_set_lag_member_attribute(
+        self.client,
+        self.lag1_member4,
+        ingress_disable=True,
+        egress_disable=True)
+
+```
+
+lag port list
+```Python
+sai_thrift_get_lag_attribute(
+                self.client, self.lag1, port_list=portlist)
+```
+
+| Steps/Cases | Goal | Expect  |
+|-|-|-|
+| Add FDB entry for port4 map to a MAC. Create lag and add port4 as a member. | Create lag and member| lag, and member created|
+| Create VLAN and add VLAN member with port0 and port1.| Prepare to send from lag to VLAN.| Vlan and members have been created.|
+| Send packet on port1 with target mac on port4. | Forwarding from port1 to port4.| Receive packet on port4.|
+| Disable egress and ingress on lag member4. send packet | Packet dropped on port4| Packet drop.|
+| Enable lag egress and ingress. Send packet with VLAN tag on lag port4 with a new dest mac.|Packet flooding on VLAN members, port0 and port1.|Packet received.|
+
```

Comment:
May consider the scale case, like how many PC we can have, and how many ports we can have in PC (I think in azure up to 4?), and is that necessary to have consecutive port numbers for one PC?

Reply:
ok, the threshold and the maximum can be checked and should be controlled from the switch aspect (another sai module), those functions(not only PC, also including FDB, ACL, VLAN volume, etc ) will be tested in that component.